lib/number.js

```diff
@@ -51,66 +51,76 @@ class Number {
-                  );
-                  m[3] = m[2].replace(/[^A-Za-z0-9]/g, "");
-                }
-                this.counter[m[1]] = (this.counter[m[1]] || 0) + 1;
-                for (var subc in this.counter)
-                  if (subc.endsWith("sub" + m[1])) this.counter[subc] = 0;
-                this.number[this.match] = this.secno(m[1]);
-                for (var supc = m[1]; supc.startsWith("sub"); )
-                  this.number[this.match] =
-                    this.secno((supc = supc.substring(3))) +
-                    "." +
-                    this.number[this.match];
-                if (m[1].endsWith("isec")) {
-                  m[3] += this.number[this.match];
-                  this.number[this.match] = "i" + this.number[this.match];
-                }
-                if (m[1].endsWith("sec")) {
-                  var s = this.number[this.match];
-                  this.toc[s] = {
-                    number: s,
-                    name: m[2],
-                    href: m[3],
-                    sub: [],
-                  };
-                  var p = s.substring(0, s.lastIndexOf("."));
-                  if (p === "" && m[1].endsWith("isec")) p = m[1];
-                  this.toc[p]?.sub.push(this.toc[s]);
-                }
-                if (m[2]) {
-                  this.refs[m[3]] = this.number[this.match];
-                  this.number[this.match] = `<a name="${m[3]}" href="#${
-                    m[3]
-                  }">${
-                    this.number[this.match].startsWith("i")
-                      ? ""
-                      : this.number[this.match]
-                  }${m[1] === "asec" ? "." : ""}</a>`;
+                  this.refs[m[1]] = true;
+                m = line.match(/ ##([A-Za-z0-9.]+)(_([A-Za-z0-9]+))?/);
+                if (m && line[m.index + m[0].length] !== "]") {
+                  if (
+                    !m[1].startsWith("sub") &&
+                    m[1].endsWith("isec") &&
+                    !this.toc[m[1]]
+                  )
+                    this.toc[m[1]] = { sub: [] };
+                  if (m[1].endsWith("sec")) {
+                    m = line.match(
+                      / ##([A-Za-z0-9.]+)(?:_[A-Za-z0-9]+)?\s+(.+)$/,
+                    );
+                    m[3] = m[2].replace(/[^A-Za-z0-9]/g, "");
+                  }
+                  this.counter[m[1]] = (this.counter[m[1]] || 0) + 1;
+                  for (var subc in this.counter)
+                    if (subc.endsWith("sub" + m[1])) this.counter[subc] = 0;
+                  this.number[this.match] = this.secno(m[1]);
+                  for (var supc = m[1]; supc.startsWith("sub"); )
+                    this.number[this.match] =
+                      this.secno((supc = supc.substring(3))) +
+                      "." +
+                      this.number[this.match];
+                  if (m[1].endsWith("isec")) {
+                    m[3] += "." + this.number[this.match];
```

Comment:
This is the main change of this PR: Insert a dot between an Appendix B entry name and the number that ensures uniqueness.